### ISSUE TITLE
some fonts are not visible

### ISSUE THREAD

**Issue by joergmschulz** [husisusi/officeonlin-install.sh#184]
under some circumstances - and this is what I am wondering about - some fonts are not rendered correctly.
This is different between two machines with /I'd say/ identical configuration. Not believing in miracles, I am still looking for the reasons.
Does anybody have an idea where to look?

Currently, it is the Helvetica font: when opened on lool server 1, the font is rendered correctly (replacing it with Arial which is installed) but on server 2, only a blank is displayed where there should be text - this happens for the identical document.

Kind regards

**Comment by ntinti** [comment]
I've upgrade to 4.0.4, 6.0-30 today and installed new MS fonts after. (https://help.nextcloud.com/t/installing-new-fonts-in-collabora/22758)

These MS fonts are shown in the font menu, but if formatting text with "arial" or any other ms font it whipes out  the text. After deinstalling the fonts the text is there again...

Maybe the fonts have to be in /usr/share/fonts/truetype _before_ installing officeonlin-install.sh:
https://help.nextcloud.com/t/i-want-to-prevent-design-collapse-by-font-in-collabora/22676/11

Updating the systemplate with  "su lool --shell=/bin/sh -c “loolwsd-systemplate-setup /opt/lool/systemplate /opt/libreoffice" does not work. Don't know if some changes should be seen in "/opt/lool/systemplate/etc/fonts/conf.d"

**Comment by Ark74** [comment]
I guess you must rebuild in order to get the fonts rendered on the server.

**Comment by ntinti** [comment]
Maybe. But a reason must be the fonts itself. 
"`apt-get install ttf-mscorefonts-installer`" installs the fonts in "`/usr/share/fonts/truetype/msttcorefonts`". 
After doing a 
'`su lool --shell=/bin/sh -c "loolwsd-systemplate-setup /opt/lool/systemplate /opt/libreoffice"`' 
these fonts are installed in "`/opt/lool/systemplate/usr/share/fonts/truetype/msttcorefonts`". 
Then I copied them to another directory `"/opt/libreoffice/instdir/share/fonts/truetype/local"`.
The effect is that some fonts can be used - other not. HaHa - Verdana works. Arial not. Then I delete all Arial instances and copied the Arial fonts from my Macbook. Voila. I can use Arial in Collabora. I think I will take all MS fonts from my Mac...

**Comment by Ark74** [comment]
Good to know, maybe there is an issue with badly encoded fonts.

**Comment by joergmschulz** [comment]
Good news, I resolved this one last night. I had many fonts installed on the system, obviously one of them acted as invalid replacement of Helvetica. Because I removed about 100 packages containing fonts, I can't say which one.
After service loolwsd stop/start the additional fonts are removed properly from lool, and, better still, Helvetica formatted texts are visible again and replaced with a proper font .

Lessons learned: 
- characters not visible when formatted with certain fonts --> check fonts installation, remove / add fonts until it works
- it is enough to stop/start loolwsd when fontconfiguration on host changed
